test/socket.io.js

```diff
@@ -1546,9 +1546,7 @@ describe('socket.io', function(){
       var srv = http();
       var sio = io(srv);
       srv.listen(function() {
-        var addr = srv.listen().address();
-        var url = 'ws://localhost:' + addr.port + '?key1=1&key2=2';
-        var socket = ioc(url);
+        var socket = client(srv, {query: {key1: 1, key2: 2}});
```

Comment:
https://github.com/socketio/socket.io-client/blob/master/lib/index.js#L44
In here, the query is parsed from the url.
`key1=1&key2=2`
You could simply add debug code in socket.io-client to see the value of `opts.query`.
Eventually, they are parsed as the same value.